### DISCUSSION TITLE
Does the Traffic Mirroring eBPF support GRE tunnels

### DISCUSSION THREAD

**Discussion by ssriley** [l3af-project/eBPF-Package-Repository#76] [Q&A]
I am trying to use the traffic mirroring example you have. I have setup the gue encapsulated tunnel and everything gets mirrored over to the collector machine i have. The problem is the software i am using to collect the network packets doesn't seem to be able to extract the packets on the tunnel interface. The monitoring software does seem to support GRE encapsulated tunnels. I am using Arkime to capture the packets on the collector. Can the traffic mirror eBPF use GRE tunnels? I can't seem to get it to mirror packets across the tunnel. I have setup a layer 2 GRETAP interface on both machines.

**Comment by sanfern** [comment]
@ssriley Thank you for exploring our traffic mirroring eBPF program, which has been specifically designed to be compatible with GUE tunnel. In order to remove this limitation, some modifications to the code will be required. Here is a https://github.com/l3af-project/eBPF-Package-Repository/issues/77 for tracking.
